snippets/javascript/password-rule-checker.md

```diff
@@ -0,0 +1,161 @@
+                readonly
+              />
+            </div>
+            <div class="form-group">
+              <label for="password">Password</label>
+              <input
+                type="password"
+                class="form-control"
+                id="password"
+                autocomplete="off"
+                placeholder="Enter your password"
+              />
+            </div>
+            <button type="submit" class="btn btn-primary m-3">Submit</button>
+            <hr />
+            <progress
+              min="0"
+              max="100"
+              value="0"
+              class="form-control"
+              id="progressbar"
+            >
+              Password
+            </progress>
+            <hr />
+            <div>
+              <small>PASSWORD SHOULD</small><br />
+              <small class="text-center" id="passlength"> be between 4-10</small
+              ><br />
+              <small class="text-center" id="smallLetters">
+                contain at least 1 lower case </small
+              ><br />
+              <small class="text-center" id="capitalLetters">
+                contain at least 1 upper case </small
+              ><br />
+              <small class="text-center" id="numbers">
+                contain at least 1 number </small
+              ><br />
+              <small class="text-center" id="specialCharacters">
+                contain at least one of *$@^#&!</small
+              >
+            </div>
+          </form>
+        </div>
+      </div>
+    </div>
+    <script src="script.js"></script>
+  </body>
+</html>
+```
+
+```js
+// Select elements using their id
+let passwd = document.getElementById("password");
+let passLength = document.getElementById("passlength");
+let sLetters = document.getElementById("smallLetters");
+let cLetters = document.getElementById("capitalLetters");
+let numbers = document.getElementById("numbers");
+let specialChars = document.getElementById("specialCharacters");
+
+// fire event for every key up
+passwd.addEventListener("keyup", function () {
+  passwordRuleChecker(passwd.value);
+});
+
+// function to check password rules
+function passwordRuleChecker(password) {
+  let passStrength = 0;
+  let progressbar = document.getElementById("progressbar");
+  if (password.match(/[a-z]+/)) {
+    passStrength += 1;
+    sLetters.className = "text-success";
+  } else {
+    sLetters.className = "text-danger";
+  }
+  if (password.match(/[A-Z]+/)) {
+    passStrength += 1;
+    cLetters.className = "text-success";
+  } else {
+    cLetters.className = "text-danger";
+  }
+  if (password.match(/[0-9]+/)) {
+    passStrength += 1;
+    numbers.className = "text-success";
+  } else {
+    numbers.className = "text-danger";
+  }
+  if (password.match(/[*$@^#&!]+/)) {
+    passStrength += 1;
+    specialChars.className = "text-success";
+  } else {
+    specialChars.className = "text-danger";
+  }
+
+  if (password.length < 4 || password.length > 10) {
```

Comment:
In my opinion, this would result in a too short password. If you're planning to use this in a real-world-application, I would suggest using a password with a minimal length of at least 8 and a maximum length of 100 (things as password generators do exist which usually generate passwords that are about 16/32/64 characters.